criu/files-reg.c

```diff
@@ -1384,30 +1384,43 @@ static int get_build_id_32(Elf32_Ehdr *file_header, unsigned char **build_id,
 	 * If the file has a build-id, it will be in the PT_NOTE program header
 	 * entry AKA the note sections.
 	 */
-	while (num_iterations-- && program_header <= program_header_end &&
-			program_header->p_type != PT_NOTE)
-		program_header++;
-
-	if (!num_iterations || program_header >= program_header_end) {
-		pr_warn("Couldn't find the note program header for file with fd %d\n", fd);
-		return -1;
-	}
+	while (num_iterations-- && program_header <= program_header_end) {
```

Comment:
If e_phnum is 1, this loop will do two cycles, what is wrong, isn't it?

```
avagin@avagin-glaptop:~$ cat test.c
#include <stdio.h>
#include <unistd.h>

int main()
{
	int num_iters = 2;

	while (num_iters--) {
		printf("%d\n", num_iters);
	}
	return 0;
}
avagin@avagin-glaptop:~$ gcc test.c
avagin@avagin-glaptop:~$ ./a.out 
1
0
```

Reply:
Yeah, you're right. I've changed this.